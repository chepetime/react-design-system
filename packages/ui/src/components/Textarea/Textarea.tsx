import React from "react";
import classnames from "classnames";
import css from "./Textarea.module.scss";

export interface TextareaProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Textarea = React.forwardRef<HTMLDivElement, TextareaProps>(
  (props: TextareaProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Textarea)} ref={ref}>
        <p>Textarea component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Textarea.displayName = "Textarea";
