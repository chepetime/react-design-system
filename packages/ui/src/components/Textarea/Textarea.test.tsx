
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Textarea } from "./Textarea";
import { render, screen } from "@testing-library/react";

describe("Textarea", () => {
  it("renders without errors", () => {
    render(<Textarea />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Textarea>Some children</Textarea>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Textarea>Some children</Textarea>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Textarea />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
