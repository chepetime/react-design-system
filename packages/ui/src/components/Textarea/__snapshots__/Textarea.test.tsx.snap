// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea matches snapshot 1`] = `
<div>
  <div
    class="Textarea"
  >
    <p>
      Textarea component working!
    </p>
    <div />
  </div>
</div>
`;
